mattermost/mattermost-webapp
# [MM-37064] - Onboarding flow is showing for existing users

#### Summary
This PR addresses the following issues with the onboarding process:
- Because onboarding was recently opened for self managed accounts, existing users were seeing the onboarding flow as well. In this PR we check if a user account has existed for some time (5 days) and we don't show the onboarding flow. They can instead access it from the main menu
- If onboarding is not showing, switching between threads view and channel view will not show it
- Switching between threads and channel view hides the onboarding flow. Also switching between channels hides the onboarding flow. The user can always access it from the main menu

#### Ticket Link
[MM-37064](https://mattermost.atlassian.net/browse/MM-37064?atlOrigin=eyJpIjoiMGUyYmI5NzNlN2E5NDM0YzgxNWFhNjQyYTQ3MWJjMjIiLCJwIjoiaiJ9)

[MM-37106](https://mattermost.atlassian.net/browse/MM-37106?atlOrigin=eyJpIjoiNTBlMDg0NDBjNDAzNGZmMGEzMTQxNWQyOTk4ZDIwMTYiLCJwIjoiaiJ9)

[MM-37105](https://mattermost.atlassian.net/browse/MM-37105?atlOrigin=eyJpIjoiM2EwZDI4MDZhMWY3NDNkYTk4ZDIwNWY4NjQzMTQ1MzgiLCJwIjoiaiJ9)

#### QA Steps

## Test 1
- Create an instance and add some users. We need the user that we're going to test to stay on this server for 6 days (we temporarily alter code to make it shorter)
- Add several teams and enable threads from the account settings > display
- For a user that has stayed more than 6 days on the server, the onboarding should not show up

## Test 2 (can be done immediately)
- Now create a new user. and log in with that user. Add that user to all the teams on the server
- When logged in, the onboarding flow should show for this user.
- Test these scenarios. Switching between teams should hide the onboarding and it can then be got again from main menu > Getting started
- Switch to thread view and refresh. Onboarding should show there instead of threads. Switch between thread and channel for the current team. The onboarding should disappear.
- Now switch to channel and refresh. Once again the onboarding shows up. Switch between channels and the onboarding should disappear.

#### Release Note
```release-note
None

```
